### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by kistitan**
Clearing database in case of incompatible version and logging crashes explicitly

### CODE REVIEW

File: FetLife/.idea/gradle.xml [hunk 1]
```diff
@@ -5,8 +5,7 @@
       <GradleProjectSettings>
         <option name="distributionType" value="LOCAL" />
         <option name="externalProjectPath" value="$PROJECT_DIR$" />
-        <option name="gradleHome" value="$APPLICATION_HOME_DIR$/gradle/gradle-2.8" />
-        <option name="gradleJvm" value="1.7" />
+        <option name="gradleHome" value="C:\Program Files\Android\Android Studio\gradle\gradle-2.10" />
```
Comment (spilth) [inline FetLife/.idea/gradle.xml]: The hardcoded value for gradleHome seems dangerous.

How about using Gradle Wrapper? <https://docs.gradle.org/current/userguide/gradle_wrapper.html>

Comment (spilth) [inline FetLife/.idea/gradle.xml]: I noticed at the bottom of this commit that we do. So I think it's just a matter of telling IntelliJ to use it.

Generally I'm not a fan of committing IDE settings to a repo though.

Comment (kistitan) [inline FetLife/.idea/gradle.xml]: Thanks! Good point. 
I plan to add Gradle Wrapper for a while. Maybe it is the time :)
Hard coded values are indeed not nice.
However I would rather do it in a separate pl are hard coded settings are already there for a while here there are only some changes.

About idea files in the repo I personally would follow IntelliJ recommendation:

"The .idea directory contains a set of configuration files (.xml). Each file contains only a portion of configuration data pertaining to a certain functional area which is reflected in the name of a file, for example, compiler.xml, encodings.xml, modules.xml.

Almost all of the files contain information core to the project itself, such as names and locations of its component modules, compiler settings, etc. **Thus, these files may (and should) be kept under version control.** "

Comment (spilth) [inline FetLife/.idea/gradle.xml]: Separate PR sounds good.

And I'm not going to argue against checking in .idea on a project I haven't contributed to yet :-)

**Comment by spilth** [comment]
👍

**Comment by jacortinas** [comment]
:+1:

**Pull request merged**